Model/SourcesRepository.php

```diff
@@ -0,0 +1,119 @@
+<?php
+
+namespace Ampersand\DisableStockReservation\Model;
+
+use Magento\Framework\Exception\LocalizedException;
+use Magento\Framework\Exception\NoSuchEntityException;
+use Magento\Framework\Exception\CouldNotSaveException;
+use Ampersand\DisableStockReservation\Model\ResourceModel\Sources;
+use Ampersand\DisableStockReservation\Api\Data\SourcesInterface;
+use Ampersand\DisableStockReservation\Api\SourcesRepositoryInterface;
+use Ampersand\DisableStockReservation\Api\Data\SourcesInterfaceFactory;
+use Ampersand\DisableStockReservation\Model\Sources as SourcesModel;
+use Ampersand\DisableStockReservation\Service\SourcesConverter;
+use Magento\InventorySourceSelection\Model\Result\SourceSelectionItem;
+use Magento\Framework\Model\AbstractModel;
+
+/**
+ * Class SourcesRepository
+ * @package Ampersand\DisableStockReservation\Model
+ */
+class SourcesRepository implements SourcesRepositoryInterface
+{
+    /**
+     * @var SourcesInterfaceFactory
+     */
+    protected $sourcesFactory;
+
+    /**
+     * @var Sources
+     */
+    protected $sourcesResourceModel;
+
+    /**
+     * @var SourcesConverter
+     */
+    private $sourcesConverter;
+
+    /**
+     * SourcesRepository constructor.
+     * @param SourcesInterfaceFactory $sourcesFactory
+     * @param Sources $sourcesResourceModel
+     * @param SourcesConverter $sourcesConverter
+     */
+    public function __construct(
+        SourcesInterfaceFactory $sourcesFactory,
+        Sources $sourcesResourceModel,
+        SourcesConverter $sourcesConverter
+    ) {
+        $this->sourcesFactory = $sourcesFactory;
+        $this->sourcesResourceModel = $sourcesResourceModel;
+        $this->sourcesConverter = $sourcesConverter;
+    }
+
+    /**
+     * @param string $orderId
+     *
+     * @return SourcesInterface
+     * @throws \Exception
+     */
+    public function getByOrderId(string $orderId): SourcesInterface
+    {
+        /** @var SourcesModel $sourcesModel */
+        $sourcesModel = $this->sourcesFactory->create();
+
+        $this->sourcesResourceModel->load(
+            $sourcesModel,
+            $orderId,
+            'order_id'
+        );
+
+        if (!$sourcesModel->getId()) {
+            throw new NoSuchEntityException(__('Source model with the order ID "%1', $orderId));
```

Comment:
Can we fix/update the exception message here? It's missing the last `"`... Also can we be more descriptive? So:
```
Source model with the order ID "%1" does not exist
```
Would make things a little less ambiguous